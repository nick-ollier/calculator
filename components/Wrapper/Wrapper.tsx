import styled from 'styled-components';

const StyledWrapper = styled.div<{ portrait: boolean }>`
    height: ${({ portrait }) => (portrait ? '700px' : '350px')};
    position: relative;
    width: ${({ portrait }) => (portrait ? '350px' : '700px')};
`;

const Screen = styled.div<{ darkMode: boolean }>`
    background-color: ${({ darkMode }) => (darkMode ? 'black' : 'white')};
    border-radius: 30px;
    box-shadow: 0px 0px 20px 20px rgba(0, 0, 0, 0.75);
    width: 95%;
    height: 95%;
    top: 2.5%;
    left: 2.5%;
    position: absolute;
`;

const IPhone = styled.div<{ portrait: boolean }>`
    height: 100%;
    overflow: hidden;
    position: absolute;
    width: 100%;

    &::after {
        background-image: url('/iPhone.png');
        background-position: center;
        background-repeat: no-repeat;
        background-size: contain;
        content: '';
        position: absolute;

        ${({ portrait }) =>
            portrait
                ? `
            height: 100%;
            width: 100%;
          `
                : `
          height: 200%;
          width: 200%;
          top: -50%;
          left: -50%;
          transform: rotate(-90deg);
    `}
    }
`;

const Content = styled.div<{ portrait: boolean }>`
    height: 100%;
    padding: ${({ portrait }) => (portrait ? '30px' : '20px 40px')};
    position: relative;
`;

const Overlay = styled.div<{ portrait: boolean }>`
    height: ${({ portrait }) => (portrait ? '94%' : '86%;')};
    width: ${({ portrait }) => (portrait ? '86%' : '94%;')};
    position: absolute;
    top: ${({ portrait }) => (portrait ? '3%' : '7%;')};
    left: ${({ portrait }) => (portrait ? '7%' : '3%;')};
    background: black;
    opacity: 0.75;
    border-radius: 35px;
`;

export interface WrapperProps {
    darkMode: boolean;
    portrait: boolean;
    overlay: boolean;
}

const Wrapper: React.FC<WrapperProps> = ({
    children,
    darkMode,
    portrait,
    overlay
}) => (
    <StyledWrapper portrait={portrait}>
        <Screen darkMode={darkMode} />
        <IPhone portrait={portrait} />
        <Content portrait={portrait}>{children}</Content>
        {overlay && <Overlay portrait={portrait} data-testid='overlay' />}
    </StyledWrapper>
);

export default Wrapper;
