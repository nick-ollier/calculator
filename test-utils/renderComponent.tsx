import CalculatorProvider from '@context/CalculatorContext';
import { render } from '@testing-library/react';

const renderWrapper = (ui: React.ReactElement) => {
    const WithProviders = ({ children }) => {
        return <CalculatorProvider>{children}</CalculatorProvider>;
    };

    return render(ui, { wrapper: WithProviders });
};

export * from '@testing-library/react';
export { renderWrapper as render };
