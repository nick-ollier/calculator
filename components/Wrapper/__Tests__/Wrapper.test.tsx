import { render } from '@testUtils/renderComponent';
import 'jest-styled-components';
import Wrapper, { WrapperProps } from '@components/Wrapper/Wrapper';

const renderComponent = (props?: Partial<WrapperProps>) => {
    const utils = render(
        <Wrapper portrait darkMode={false} overlay={false} {...props} />
    );

    return {
        ...utils
    };
};

describe('component: wrapper', () => {
    describe('portrait', () => {
        it('renders', () => {
            const { container } = renderComponent();
            expect(container);
        });

        it('renders with overlay', () => {
            const { container } = renderComponent({ overlay: true });
            expect(container);
        });

        it('matches snapshot', () => {
            const { asFragment } = renderComponent();
            expect(asFragment()).toMatchSnapshot();
        });

        it('matches snapshot with overlay', () => {
            const { asFragment } = renderComponent({ overlay: true });
            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe('landscape', () => {
        it('renders', () => {
            const { container } = renderComponent({ portrait: false });
            expect(container);
        });

        it('renders with overlay', () => {
            const { container } = renderComponent({
                portrait: false,
                overlay: true
            });
            expect(container);
        });

        it('matches snapshot', () => {
            const { asFragment } = renderComponent({ portrait: false });
            expect(asFragment()).toMatchSnapshot();
        });

        it('matches snapshot with overlay', () => {
            const { asFragment } = renderComponent({
                portrait: false,
                overlay: true
            });
            expect(asFragment()).toMatchSnapshot();
        });
    });

    describe('with darkmode', () => {
        it('renders', () => {
            const { container } = renderComponent({ darkMode: true });
            expect(container);
        });

        it('matches snapshot', () => {
            const { asFragment } = renderComponent({ darkMode: true });
            expect(asFragment()).toMatchSnapshot();
        });
    });
});
