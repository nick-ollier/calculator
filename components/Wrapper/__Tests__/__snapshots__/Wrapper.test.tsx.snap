// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: wrapper landscape matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: 350px;
  position: relative;
  width: 700px;
}

.c1 {
  background-color: white;
  border-radius: 30px;
  box-shadow: 0px 0px 20px 20px rgba(0,0,0,0.75);
  width: 95%;
  height: 95%;
  top: 2.5%;
  left: 2.5%;
  position: absolute;
}

.c2 {
  height: 100%;
  overflow: hidden;
  position: absolute;
  width: 100%;
}

.c2::after {
  background-image: url('/iPhone.png');
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  content: '';
  position: absolute;
  height: 200%;
  width: 200%;
  top: -50%;
  left: -50%;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c3 {
  height: 100%;
  padding: 20px 40px;
  position: relative;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    />
    <div
      class="c3"
    />
  </div>
</DocumentFragment>
`;

exports[`component: wrapper landscape matches snapshot with overlay 1`] = `
<DocumentFragment>
  .c0 {
  height: 350px;
  position: relative;
  width: 700px;
}

.c1 {
  background-color: white;
  border-radius: 30px;
  box-shadow: 0px 0px 20px 20px rgba(0,0,0,0.75);
  width: 95%;
  height: 95%;
  top: 2.5%;
  left: 2.5%;
  position: absolute;
}

.c2 {
  height: 100%;
  overflow: hidden;
  position: absolute;
  width: 100%;
}

.c2::after {
  background-image: url('/iPhone.png');
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  content: '';
  position: absolute;
  height: 200%;
  width: 200%;
  top: -50%;
  left: -50%;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c3 {
  height: 100%;
  padding: 20px 40px;
  position: relative;
}

.c4 {
  height: 86%;
  width: 94%;
  position: absolute;
  top: 7%;
  left: 3%;
  background: black;
  opacity: 0.75;
  border-radius: 35px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    />
    <div
      class="c3"
    />
    <div
      class="c4"
      data-testid="overlay"
    />
  </div>
</DocumentFragment>
`;

exports[`component: wrapper portrait matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: 700px;
  position: relative;
  width: 350px;
}

.c1 {
  background-color: white;
  border-radius: 30px;
  box-shadow: 0px 0px 20px 20px rgba(0,0,0,0.75);
  width: 95%;
  height: 95%;
  top: 2.5%;
  left: 2.5%;
  position: absolute;
}

.c2 {
  height: 100%;
  overflow: hidden;
  position: absolute;
  width: 100%;
}

.c2::after {
  background-image: url('/iPhone.png');
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  content: '';
  position: absolute;
  height: 100%;
  width: 100%;
}

.c3 {
  height: 100%;
  padding: 30px;
  position: relative;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    />
    <div
      class="c3"
    />
  </div>
</DocumentFragment>
`;

exports[`component: wrapper portrait matches snapshot with overlay 1`] = `
<DocumentFragment>
  .c0 {
  height: 700px;
  position: relative;
  width: 350px;
}

.c1 {
  background-color: white;
  border-radius: 30px;
  box-shadow: 0px 0px 20px 20px rgba(0,0,0,0.75);
  width: 95%;
  height: 95%;
  top: 2.5%;
  left: 2.5%;
  position: absolute;
}

.c2 {
  height: 100%;
  overflow: hidden;
  position: absolute;
  width: 100%;
}

.c2::after {
  background-image: url('/iPhone.png');
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  content: '';
  position: absolute;
  height: 100%;
  width: 100%;
}

.c3 {
  height: 100%;
  padding: 30px;
  position: relative;
}

.c4 {
  height: 94%;
  width: 86%;
  position: absolute;
  top: 3%;
  left: 7%;
  background: black;
  opacity: 0.75;
  border-radius: 35px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    />
    <div
      class="c3"
    />
    <div
      class="c4"
      data-testid="overlay"
    />
  </div>
</DocumentFragment>
`;

exports[`component: wrapper with darkmode matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: 700px;
  position: relative;
  width: 350px;
}

.c1 {
  background-color: black;
  border-radius: 30px;
  box-shadow: 0px 0px 20px 20px rgba(0,0,0,0.75);
  width: 95%;
  height: 95%;
  top: 2.5%;
  left: 2.5%;
  position: absolute;
}

.c2 {
  height: 100%;
  overflow: hidden;
  position: absolute;
  width: 100%;
}

.c2::after {
  background-image: url('/iPhone.png');
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  content: '';
  position: absolute;
  height: 100%;
  width: 100%;
}

.c3 {
  height: 100%;
  padding: 30px;
  position: relative;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    />
    <div
      class="c3"
    />
  </div>
</DocumentFragment>
`;
